### ISSUE TITLE
Follow up for design review and clean up of TODO items on ValueSet selection page

### ISSUE THREAD

**Issue by katyasoup** [CDCgov/dibbs-query-connector#210]
In order to unblock [QUE-74](https://linear.app/skylight-cdc/issue/QUE-74/search-from-the-valueset-selection-page) and [QUE-69](https://linear.app/skylight-cdc/issue/QUE-69/render-the-side-panel-on-concept-selection), let's merge the ValueSet page work and address any small changes in this follow-up ticket

* Any TODO items from [QUE-65](https://linear.app/skylight-cdc/issue/QUE-65/create-the-valueset-selection-page) are addressed
  * Bug: fix arrow orientation on Accordion headers when clicking between conditions
  * Cleanup: streamline logic for backnav button and its conditional text
* Any items found in design review
